llvm/lib/Target/LoongArch/LoongArchFloat32InstrInfo.td

```diff
@@ -208,16 +212,18 @@ def : PatFPSetcc<SETUO,  FCMP_CUN_S,  FPR32>;
 def : PatFPSetcc<SETLT,  FCMP_CLT_S,  FPR32>;
 
 multiclass PatFPBrcond<CondCode cc, LAInst CmpInst, RegisterClass RegTy> {
-  def : Pat<(brcond (xor (GRLenVT (setcc RegTy:$fj, RegTy:$fk, cc)), -1),
-                     bb:$imm21),
+  def : Pat<(loongarch_brcond (NotBoolXor (GRLenVT (setcc RegTy:$fj, RegTy:$fk, cc))),
+                              bb:$imm21),
             (BCEQZ (CmpInst RegTy:$fj, RegTy:$fk), bb:$imm21)>;
-  def : Pat<(brcond (GRLenVT (setcc RegTy:$fj, RegTy:$fk, cc)), bb:$imm21),
+  def : Pat<(loongarch_brcond (GRLenVT (setcc RegTy:$fj, RegTy:$fk, cc)), bb:$imm21),
             (BCNEZ (CmpInst RegTy:$fj, RegTy:$fk), bb:$imm21)>;
 }
 
 defm : PatFPBrcond<SETOEQ, FCMP_CEQ_S, FPR32>;
+defm : PatFPBrcond<SETEQ , FCMP_CEQ_S, FPR32>;
```

Comment:
I'm not sure whether SETEQ and SETLE can be used for by FCmpInst.

https://github.com/llvm/llvm-project/blob/main/llvm/include/llvm/Target/TargetSelectionDAG.td#L922-L926
https://github.com/llvm/llvm-project/blob/main/llvm/include/llvm/IR/InstrTypes.h#L670-L673

Reply:
I remember this was added after encountering a "cannot select instruction" error.

Reply:
https://github.com/llvm/llvm-project/blob/e1ac57c1a560b1d9891f93081e2f9c862c4a8d77/llvm/lib/Target/LoongArch/LoongArchFloat32InstrInfo.td#L197
https://github.com/llvm/llvm-project/blob/e1ac57c1a560b1d9891f93081e2f9c862c4a8d77/llvm/lib/Target/LoongArch/LoongArchFloat32InstrInfo.td#L200

Reply:
https://github.com/llvm/llvm-project/blob/8c6559193f1d7bd636167b2d54c90e8c49f22f70/llvm/include/llvm/CodeGen/ISDOpcodes.h#L1679-L1681